homeassistant/components/shelly/__init__.py

```diff
@@ -307,6 +309,15 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if not entry.data.get(CONF_SLEEP_PERIOD):
         platforms = RPC_PLATFORMS
 
+    # delete push update issue if it exists
+    if get_device_entry_gen(entry) == 1:
+        LOGGER.debug(
+            "Deleting issue %s", PUSH_UPDATE_ISSUE_ID.format(unique=entry.unique_id)
+        )
+        ir.async_delete_issue(
+            hass, DOMAIN, PUSH_UPDATE_ISSUE_ID.format(unique=entry.unique_id)
+        )
+
```

Comment:
we can move this to line 337 (or after) to eliminate the if

Reply:
Good point

Reply:
Done